cmd/share.go

```diff
@@ -103,8 +106,16 @@ var shareCmd = &cobra.Command{
 
 		    log.Debugf("Processing chunk '%d'...", i_chunk)
 
+			// Check if a previous goroutine is running to refresh the session for
+			// federated identity. If so, stop it before starting a new one.
+			federated_identity.StopRefreshSession()
```

Comment:
should this be called outside of the `if` block regardless the latter's truth value? I.e. what happens if we stop refreshing without starting a new session?

Reply:
I think it should be fine because it checks if a goroutine is currently running before trying to stop anything. In the case where `AwsRoleArn` isn't used, this should just continue but I can move it to before `StartRefreshSession` in the if-block if that makes more sense

Reply:
I'm a bit confused about why we need to call stop here at all? can't we just call start once if `opts.AwsRoleArn != ""` and then be set for the whole process?

Reply:
For each chunk, a new session object is retrieved so the refresh from the previous chunk would be refreshing a session that isn't used anymore. In large batches where there are many chunks, we should stop the refresh on the previous one so we don't have multiple concurrent goroutines running when only one is needed at a time. I'm a novice at Golang though so if I'm misunderstanding how any of this works, let me know!

Reply:
right but why do we create a new session every chunk? also i'm realizing that you didn't actually write that line of code, so totally valid to say we just want to keep the behavior the same there for this PR

Reply:
Oh I see, yeah that's a totally fair question. I didn't want to change much of the logic outside of adding the impersonation but we should consider updating that. I create another [ticket](https://tempuslabs.atlassian.net/browse/EDT-751?atlOrigin=eyJpIjoiNTQxYmJiZmI2MTliNGUzNmFjN2JmM2NmMzI2YTYyNWEiLCJwIjoiaiJ9) to handle another issue with how these sessions and clients are handled because right now it doesn't support cross-region resource access either. Happy to follow up and explore how to do this without recreating the session once I get my more urgent tickets done!